Owncloud iOs Client/Tabs/SharedTap/SharedViewController.m

```diff
@@ -1198,6 +1175,49 @@ - (void)swipeableTableViewCell:(SWTableViewCell *)cell didTriggerLeftUtilityButt
     }
 }
 
+- (void) shareOptionSelectedForFile:(FileDto *)file {
+    DLog(@"Share Link Option");
+    
+    if (file) {
+        
+        ShareMainViewController *share = [[ShareMainViewController alloc] initWithFileDto:file];
+        OCNavigationController *nav = [[OCNavigationController alloc] initWithRootViewController:share];
+        
+        if (IS_IPHONE) {
+            [self presentViewController:nav animated:YES completion:nil];
+        } else {
+            AppDelegate *app = (AppDelegate*)[[UIApplication sharedApplication] delegate];
+            nav.modalPresentationStyle = UIModalPresentationFormSheet;
+            [app.splitViewController presentViewController:nav animated:YES completion:nil];
+        }
+        
+    }else{
+        [self showError:NSLocalizedString(@"default_not_possible_msg", nil)];
+    }
+
+}
+
+- (void) unShareLinkOptionSelectedForRemoteFileId:(NSInteger)remoteFileId {
+    
+    DLog(@"Unshare button pressed");
+    
+    if (_mShareFileOrFolder) {
+        //Create new object
```

Comment:
Here you aren't creating a new object, in fact you are assigning nil value to a global variable. Delete the comment or do what the comment say. It's confusing.

Reply:
👍 Removed wrong old comments and update other